src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.unit.cs

```diff
@@ -237,13 +234,10 @@ public void FileSystemWatcher_Filter()
             watcher.Filter = "abc.dll";
             Assert.Equal("abc.dll", watcher.Filter);
 
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || // expect no change for OrdinalIgnoreCase-equal strings
-                RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            if (!(PlatformDetection.IsFullFramework || PlatformDetection.IsOSX))
             {
-                // expect no change for OrdinalIgnoreCase-equal strings
-                // it's unclear why desktop does this but preserve it for compat
                 watcher.Filter = "ABC.DLL";
-                Assert.Equal("abc.dll", watcher.Filter);
+                Assert.Equal("ABC.DLL", watcher.Filter);
```

Comment:
While this existing behavior is somewhat surprising, there isn't a strong reason to introduce a potential break here. I'd lean towards making it match. @pjanotti, what do you think?